docs/features/shared-volumes-with-NFS.md

```diff
@@ -0,0 +1,129 @@
+1. docker volume rm - the volume must be qualified by the remote dir label as below,
+   1. _docker volume rm vol-name@remote-dir-label_
+
+
+# Extending the vSphere driver to support shared volumes
+
+The vSphere plugin driver supports an extendable framework that allows adding support for volumes backed by specific storage types. Presently the driver supports VMDK backed volumes on storage available to the local ESX host. A new module to support NFS backed volumes (on local or NFS mounted directories) is added to the existing framework that supports all of the docker volume functions as listed above.
+
+The new NFS backed volumes will support NFS v3/v4.0 protocols to mount and use remote directories to host docker volumes. Docker volumes will be supported as folders on the mounted directories and exported to containers in the same way that VMDK volumes are made available to containers.
+
+# Plugin configuration for remote directories
+
+The plugin configuration defined in /etc/docker-volume-vsphere.conf must be modified to include the details of the remote directories to use to locate volumes. The format describing a remote directory is as shown below,
+
+## /etc/docker-volume-vsphere.conf
+{
+"MaxLogAgeDays": 28,
+"MaxLogSizeMb": 100,
+"LogPath": "/var/log/docker-volume-vsphere.log",
+"LogLevel": "debug",
+_“RemoteDirs”: {_
+     _"DefaultLabel: remote-dir-label"_
+     _"RemoteDirTbl: {"_
+        _“<dir-A-label>” :   {_
+             _“Path”: <remote path to mount>,_
+             _“Args: <mount args>,_
+             _“Type”: <FS type – nfs, nfs4>,_
+             _“VolPath”: <optional path name under path above to locate volumes>,_
+	     _"Src": <label of another entry in this list to use to locate volumes>_
+        _},_
+        _“<dir-B-label>” :   {_
+             _“Path”: <remote path to mount>,_
+             _“Args: <mount args>,_
+             _“Type”: <FS type – nfs, nfs4>,_
+             _“VolPath”: <optional path name under path above to locate volumes>,_
+	     _"Src": <label of another entry in this list to use to locate volumes>_
+        _},_
+        _“<dir-C-label>” :   {_
+             _“VolPath”: <optional path name under path above to locate volumes>_
+	     _"Src": "dir-A-label"_
+        _},_
+     _},_
+  _}_
+
+The configuration items are described below:
+
+1. “RemoteDirs”:  Map of remote dirs.
+1. “<dir-A-label>” – must be unique and admin defined and identifies a specific remote dir or another entry within the list
+1. “addr” – IP or fully qualified host name of server
+1. “args – args to mount remote dir.
+1. “path” – path on server to mount
+1. “type” – FS type of remote dir.
+1. “vol-path” – a path name that is relative to the mounted path abnove, volumes will be create relative to this path if non-null, if not specified then volumes are located under path above. The _vol-path_ entry allows the admin  admin to separate volumes for different docker service stacks within a single remote dir.
```

Comment:
OK, its a flexibility for the user to have simpler definitions of the locations where they want to place their volumes vs. repeating the same remote dir config multiple times.

Remote-dir is the remote folder to mount - like <host addr>:<folder path>
VolPath - suffixes this path to _folder path_ above so user has control where the volumes get placed. The remote folder need not necessarily be used only for docker volumes - its like a tenant-folder in VMDK backed volumes.
Src - allows user to avoid repeating a remote folder definition in the config file. As mentioned in the doc, the remote directories are placed in a table in the config file and an entry can refer another entry by specifying that entry's label as a _Src_. Its only a flexibility to the user.